packages/ra-ui-materialui/src/list/datagrid/DatagridRow.tsx

```diff
@@ -114,7 +114,11 @@ const DatagridRow: FC<DatagridRowProps> = React.forwardRef((props, ref) => {
                 return;
             }
             if (['edit', 'show'].includes(type)) {
-                navigate(createPath({ resource, id, type }));
+                if(event.ctrlKey || event.metaKey) {
+                    window.open(`#${createPath({ resource, id, type })}`);
```

Comment:
This only works if the app uses the HashRouter, and will fail in other cases.

Reply:
@fzaninotto Could you please give me some hints on how to deal with it?

Reply:
Sorry, I know no easy way to do this, you'll have to search for it.